File: 전예진/Stack&Queue/prob2.py
from math import ceil

def solution(progresses, speeds):
    answer = []

    time = []
    for i in range(len(progresses)):
        time.append(ceil((100 - progresses[i]) / speeds[i]))

    stack = []   
    for i in range(len(time)):
        while stack and time[i] > time[stack[0]]:
            top = stack.pop()
        
            if not stack:
                answer.append(i - top)
        
        stack.append(i)
    
    if stack:
        answer.append(len(stack))
    
    return answer

File: 이주은/Stack-Queue/prob2.java
//[프로그래머스] 기능개발 (https://school.programmers.co.kr/learn/courses/30/lessons/42586)

import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> list = new ArrayList<>();
        int[] compDays = new int[progresses.length];
        
        //완료 일수 계산
        for(int i=0; i<progresses.length; i++){
            double daysDouble =  (100 - progresses[i])/(double)speeds[i];
            compDays[i] = (int) Math.ceil(daysDouble);
            System.out.println(compDays[i]);
        }
        
        for(int i=0; i<progresses.length; i++){
            int cnt = 1;
            for(int j=i+1; j<progresses.length; j++){
                if(compDays[i] >= compDays[j])
                    cnt++;
                else
                    break;
            }
            i += cnt -1;
            list.add(cnt);
        }
        
        int[] answer = new int[list.size()];
        for(int i=0; i<answer.length; i++){
            answer[i] = list.get(i);
        }
        
        return answer;
    }
}
